qbraid/devices/aws/device.py

```diff
@@ -232,6 +233,10 @@ def run(
 
         """
         run_input, qbraid_circuit = self._compat_run_input(run_input)
+
+        if self._provider_name == "IonQ" and self._name == "Harmony":
+            run_input = braket_ionq_compilation(run_input)
```

Comment:
Here, the `braket_ionq_compilation()` conditional should actually be written into the `_compat_run_input()` method. The reason is that the compilation function already performs conversion to `pytket` and back to `braket`, so performing an extra conversion to `braket` beforehand is redundant.

Reply:
Hello,

I have changed the transpiler from qBraid to pytket-braket and the above gates are getting compiled. In addition to that, I have put the conditional inside the `_compat_run_input`, now it doesn't transpile from pytket -> braket -> pytket -> braket anymore.